lib/run.bash

```diff
@@ -23,21 +23,31 @@ compose_cleanup() {
 check_linked_containers_and_save_logs() {
   local service="$1"
   local logdir="$2"
+  local uploadlogs="$3"
+  local uploadall="false"
+
+  if [[ "$uploadlogs" =~ ^(false|off|0|never)$ ]]; then
+    # Skip all if we are not uploading logs
+    return
+  elif [[ "$uploadlogs" =~ ^(true|on|1|always)$ ]]; then
+    uploadall="true"
+  fi
 
   [[ -d "$logdir" ]] && rm -rf "$logdir"
   mkdir -p "$logdir"
 
-  containers=$(docker_ps_by_project --format '{{.ID}}\t{{.Label "com.docker.compose.service"}}')
-  IFS=$'\n'
-  for line in ${containers} ; do
-    if [[ -z "${line}" ]]; then
+  # Get array of containers
+  mapfile -t containers < <(docker_ps_by_project --format '{{.ID}}\t{{.Label "com.docker.compose.service"}}')
```

Comment:
This will be incompatible with Bash 3.x

Reply:
Is there a reason to keep bash 3.x compat? 4.x was released 10 years ago and no LTS os seems to use it.

As far as I could check quickly using docker:
- Alpine: 5.0
- Ubuntu 16.04: 4.3
- Debian 8: 4.3
- Debian 7: 4.2
- Centos 7: 4.2
- RHEL 6: 4.1

Reply:
Older macOS versions, we could certainly look at increasing it in future, but I'd prefer not to do that as part of this.

Reply:
It's mostly OSX that causes this:

```
bash-3.2$ bash --version
bash --version
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin18)
Copyright (C) 2007 Free Software Foundation, Inc.
```

We've dropped Bash 3.x support in this plugin before, and it caused lots of problems for OSX users. When OSX bakes in Bash 4.x then we can ease up this restriction, but until then we'll have to stick to Bash 3.x'isms… sorry @pecigonzalo!

Reply:
Ill have to revert back to while or loop and IFS then, ill do it as a separate commit and keep the history, as IMO we should use `mapfile` as its a much better string->list tool.

I can test if now it's testable altho, to be honest, I was not even aware the plugins had a requirement to be bash 3.x compatible and this was just using an unused feature.

Keeping macOS compat is prob quite hard and limiting as not only bash but many Linux tools do not exist or are different over there.

Reply:
@lox would something like `IFS=$'\n' read -r -a containers <<< <(docker_ps_by_project --format '{{.ID}}\t{{.Label "com.docker.compose.service"}}')` work for bash3?

Reply:
I think so, as its used in other files

Reply:
Im back to BATS eating the output if I use read and while. Could you check my comment on: https://github.com/buildkite-plugins/docker-compose-buildkite-plugin/pull/222#issuecomment-506295024 ?